packages/compass-e2e-tests/helpers/commands/run-find-operation.js

```diff
@@ -0,0 +1,129 @@
+    Selectors.queryBarOptionInputCollation(tabName),
+    value
+  );
+}
+
+async function setMaxTimeMS(client, tabName, value) {
+  const selector = Selectors.queryBarOptionInputMaxTimeMS(tabName);
+  await client.clickVisible(selector);
+  await client.setValue(selector, value);
+}
+
+async function setSkip(client, tabName, value) {
+  const selector = Selectors.queryBarOptionInputSkip(tabName);
+  await client.clickVisible(selector);
+  await client.setValue(selector, value);
+}
+
+async function setLimit(client, tabName, value) {
+  const selector = Selectors.queryBarOptionInputLimit(tabName);
+  await client.clickVisible(selector);
+  await client.setValue(selector, value);
+}
+
+async function runFind(client, tabName) {
+  await client.clickVisible(Selectors.queryBarApplyFilterButton(tabName));
+}
+
+async function isOptionsExpanded(client, tabName) {
+  // it doesn't look like there's some attribute on the options button or container that we can easily check, so just look for a field that exists if it is expanded
+  return await client.isVisible(Selectors.queryBarOptionInputProject(tabName));
+}
+
+async function waitUntilCollapsed(client, tabName) {
+  await client.waitForVisible(
+    Selectors.queryBarOptionInputProject(tabName),
+    1000,
+    true
+  );
+}
+
+async function collapseOptions(client, tabName) {
+  if (!(await isOptionsExpanded(client, tabName))) {
+    return;
+  }
+
+  await client.clickVisible(Selectors.queryBarOptionsToggle(tabName));
+  await waitUntilCollapsed(client, tabName);
+}
+
+async function waitUntilExpanded(client, tabName) {
+  await client.waitForVisible(Selectors.queryBarOptionInputProject(tabName));
+}
+
+async function expandOptions(client, tabName) {
+  if (await isOptionsExpanded(client, tabName)) {
+    return;
+  }
+
+  await client.clickVisible(Selectors.queryBarOptionsToggle(tabName));
+  await waitUntilExpanded(client, tabName);
+}
+
+module.exports = function (app) {
+  return async function runFindOperation(
+    tabName,
+    filter,
+    {
+      project = '',
+      sort = '',
+      maxTimeMS = '',
+      collation = '',
+      skip = '',
+      limit = '',
+    } = {}
+  ) {
+    const { client } = app;
+
+    const queryBarSelector = Selectors.queryBar(tabName);
+
+    // look up the current resultId
+    const initialResultId = await client.getAttribute(
+      queryBarSelector,
+      'data-result-id'
+    );
+
+    // now we can easily see if we get a new resultId
+    await setFilter(client, tabName, filter);
+    if (project || sort || maxTimeMS || collation || skip || limit) {
+      await expandOptions(client, tabName);
+
+      await setProject(client, tabName, project);
+      await setSort(client, tabName, sort);
+      await setMaxTimeMS(client, tabName, maxTimeMS);
+      await setCollation(client, tabName, collation);
+      await setSkip(client, tabName, skip);
+      await setLimit(client, tabName, limit);
+    } else {
+      await collapseOptions(client, tabName);
```

Comment:
This relies on the fact that the app ignores the options once you collapse them even though the values are still there in the form fields. So I don't have to clear them out manually if they are still set and we don't want to specify them.

Then we can run either a simple find or a find with options like project and so on AND it doesn't care whether you've run a query before or not. Which is very useful for when you want to be able to work on tests by focusing just the test you're on and you still want it to pass if you run the whole suite.